src/parameters.jl

```diff
@@ -217,7 +219,9 @@ value(X::PositiveDefinite) = A_At(vec_to_tril(X.L))
 
 function flatten(::Type{T}, X::PositiveDefinite) where {T<:Real}
```

Comment:
This is not related to the commit, but I think most Statistics packages work with the upper triangular, if I am not mistaken?


```julia
using Distributions, LinearAlgebra
Σ1 = UpperTriangular([1. .5; .5 1.])
Σ2 = LowerTriangular([1. .5; .5 1.])
Symmetric(Σ1) # 1 0.5 0.5 1
Symmetric(Σ2) # 1 0.0 0.0 1
```

Reply:
Hmm. I agreee with your assertion that most packages use the upper triangle in the Julia ecosystem, but I'm not sure that it's a problem here, because the user should interact with the `PositiveDefinite` type via the `positive_definite` function, which just requires that the user provide a `StridedMatrix`which is positive definite (we should probably widen that to include `Symmetric` matrices...). Once inside that functionality, asking for the `L` field of a `Cholesky` is fine, albeit it may not be totally optimal.

Or maybe I've misunderstood where you're coming from with this?

Reply:
Yes, you are right. It also should probably be the job of the user to check if his transformations make sense.

Reply:
>  asking for the `L` field of a `Cholesky` is fine, albeit it may not be totally optimal.

Where "not totally optimal" in this case is "an extra copy() of the entire matrix" (https://github.com/JuliaLang/julia/pull/42920, https://github.com/JuliaStats/PDMats.jl/issues/143, note that the bugfix for that created [yet another AD bug](https://github.com/JuliaStats/PDMats.jl/pull/144#issuecomment-961008928) unfortunately)...